File: csharp/sum-of-multiples/SumOfMultiples.cs
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
      return Enumerable
        .Range(1, max-1)
        .Where(n => multiples.Any(m => m != 0 && n % m == 0))
        .Sum();
    }
}

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit: int, multiples_of: list[int]) -> int:

    # remove 0
    multiples_of = [m for m in multiples_of if m != 0]

    multiples = set()
    for n in range(1, limit):
        for m in multiples_of:
            if n % m == 0:
                multiples.add(n)
                break

    return sum(multiples)
